File: Programming/OOP/03.ExtensionMethodsDelegatesLambdaLINQ/NamesByAge/Student.cs
namespace NamesByAge
{
    public class Student
    {
        public string FirstName { get; set; }

        public string LastName { get; set; }

        public int Age { get; set; }

        public Student(string firstName, string lastName, int age)
        {
            this.FirstName = firstName;
            this.LastName = lastName;
            this.Age = age;
        }

        public override string ToString()
        {
            return string.Format("{0} {1} => age: {2}", this.FirstName, this.LastName, this.Age);
        }
    }
}

File: Web/JavaScript-Apps/01.UnderscoreJS/scripts/classes/student.js
define(['person'], function (Person) {
    'use strict';
    var Student;

    Student = (function () {
        function Student(firstName, lastName, age, marks) {
            Person.call(this, firstName, lastName, age);
            this._marks = marks;
        }

        Student.prototype = new Person();
        Student.prototype.constructor = Student;

        Student.prototype.getAllMarks = function () {
            return this._marks;
        };

        Student.prototype.getAverageScore = function () {
            var sum,
                i,
                len;

            sum = 0;
            for (i = 0, len = this.getAllMarks().length; i < len; i++) {
                sum += this.getAllMarks()[i];
            }

            return sum / len;
        };

        Student.prototype.toString = function () {
            var parent = Person.prototype.toString.call(this);

            return parent + ', average score: ' + this.getAverageScore();
        };

        return Student;
    }());

    return Student;
});